final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,285 @@
+    return parser
+
+
+def start_logging() -> None:
+    """Setting up logging basic configuration"""
+    root = logging.getLogger()
+    logging.basicConfig(
+        filename=LOG_FILE_NAME,
+        filemode='w',
+        format='[%(asctime)s] %(levelname)-8s %(message)s',
+        level=logging.DEBUG
+    )
+    if "--verbose" in sys.argv:
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setLevel(logging.DEBUG)
+        formatter = logging.Formatter('[%(asctime)s] %(levelname)-8s %(message)s')
+        handler.setFormatter(formatter)
+        root.addHandler(handler)
+
+
+def main() -> None:
+    """The main entry point of the application"""
+    try:
+        start_logging()
+        logging.info('Application started')
+
+        arg_parser = create_parser()
+        logging.info('Argument parser created')
+
+        check_the_arguments_amount()
+
+        url = find_url()
+
+        check_url_and_connection(url)
+
+        args = arg_parser.parse_args()
+
+        rss_url = args.source
+        limit = args.limit
+
+        parsed_rss = parse_rss(rss_url)
+        logging.info('RSS URL parsed')
+
+        articles_list = get_articles(parsed_rss, limit)
+        logging.info('Articles list created')
+
+        if args.json:
+            print_rss_json(articles_list)
+            logging.info('Articles was printed as json')
+        else:
+            print_rss_articles(articles_list)
+            logging.info(f'{len(articles_list)} articles was printed')
+
+        logging.info('Application ended')
+    except custom_error.UrlNotFoundInArgsError:
+        logging.error('UrlNotFoundInArgsError')
+        logging.info('Application ended')
+        print("Url is not found in the arguments")
+    except custom_error.NotEnoughArgumentsError:
+        logging.error('Not enough arguments(URL link is required)')
+        logging.info('Application ended')
+        print("Not enough arguments(URL link is required)")
+    except custom_error.NotValidUrlError:
+        logging.error('Not valid url')
+        logging.info('Application ended')
+        print('Not valid url')
+    except custom_error.ConnectionFailedError as error:
+        logging.error(error.message)
+        logging.info('Application ended')
+        print('Connection failed')
+    except custom_error.ArticleKeyError as error:
+        logging.error(error.message)
+        logging.info('Application ended')
+
+
+if __name__ == '__main__':
+    main()
+
+    # python rss_reader.py "https://news.yahoo.com/rss/"
+    # python rss_reader.py "https://news.tut.by/rss/economics.rss"
+    # python rss_reader.py 'http://rss.cnn.com/rss/edition_world.rss'
+    # python rss_reader.py http://feeds.bbci.co.uk/news/rss.xml?edition=uk
+    # python rss_reader.py "http://billmaher.hbo.libsynpro.com/rss"
+    # python rss_reader.py "https://www.craigslist.org/about/best/all/index.rss"
```

Comment:
It is better to put RSS sources you have tested in README.md instead of keeping them here